### DISCUSSION TITLE
Storage - file url after manual upload

### DISCUSSION THREAD

**Discussion by loic-hamdi** [supabase/supabase#2036] [Questions] [answered]
Hi, 
Is there an equivalent of gs:// from firestore?
I can't find a way to retrieve the url of an image uploaded manually in the storage.
Any idea please?

**Comment by inian** [comment, marked as answer]
Hey @zzterrozz, if you are looking to get the URL of the object via the dashboard, there is a `Copy URL` button on the sidebar. Or are you trying to get the URL of the asset programatically? 

We don't use a custom scheme like Firestore, the URLs are just HTTPS urls. 

![image](https://user-images.githubusercontent.com/2155545/122849856-e185af80-d33e-11eb-8dbb-346679d8892d.png)

**Reply by loic-hamdi** [reply]
Programatically of course

**Reply by inian** [reply]
For public buckets you can use the [getPublicURL](https://supabase.io/docs/reference/javascript/storage-from-getpublicurl) method and for non public buckets you can use the [createSignedURL method](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl)

**Reply by loic-hamdi** [reply]
> For public buckets you can use the [getPublicURL](https://supabase.io/docs/reference/javascript/storage-from-getpublicurl) method and for non public buckets you can use the [createSignedURL method](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl)

Excellent, I will give it a try. Thank you @inian 
Keep the great work on Supabase, we love what you guys are creating!

**Reply by inian** [reply]
Thanks @zzterrozz :)
